cmd/aws/create_test.go

```diff
@@ -0,0 +1,402 @@
+package aws
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"testing"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/service/ec2"
+	ec2Types "github.com/aws/aws-sdk-go-v2/service/ec2/types"
+	"github.com/aws/aws-sdk-go-v2/service/ssm"
+	ssmTypes "github.com/aws/aws-sdk-go-v2/service/ssm/types"
+	"github.com/stretchr/testify/assert"
+)
+
+type mockCredentialsProvider struct {
+	creds aws.Credentials
+	err   error
+}
+
+func (m *mockCredentialsProvider) Retrieve(ctx context.Context) (aws.Credentials, error) {
+	return m.creds, m.err
+}
+
+func TestValidateCredentials(t *testing.T) {
+	tests := []struct {
+		name        string
+		creds       aws.Credentials
+		err         error
+		expectedErr error
+	}{
+		{
+			name: "valid credentials",
+			creds: aws.Credentials{
+				AccessKeyID:     "test-access-key-id",
+				SecretAccessKey: "test-secret-access-key",
+				SessionToken:    "test-session-token",
+			},
+			err:         nil,
+			expectedErr: nil,
+		},
+		{
+			name:        "failed to retrieve credentials",
+			creds:       aws.Credentials{},
+			err:         errors.New("failed to retrieve credentials"),
+			expectedErr: errors.New("failed to retrieve AWS credentials: failed to retrieve credentials"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockProvider := &mockCredentialsProvider{
+				creds: tt.creds,
+				err:   tt.err,
+			}
+
+			creds, err := getSessionCredentials(context.Background(), mockProvider)
+			if tt.expectedErr != nil {
+				assert.Nil(t, creds)
+				assert.EqualError(t, err, tt.expectedErr.Error())
+			} else {
+				assert.NotNil(t, creds)
+				assert.NoError(t, err)
+				assert.Equal(t, tt.creds, *creds)
+			}
```

Comment:
Small word of advice: `assert` in `stretchr/testify/assert` fails the test, but doesn't stop it.

If you fail the assertion, and then say you need the value being not nil, this will panic your test for the wrong reason.

You almost always will want to use `stretchr/testify/assert/require` with `require.Nil`, `require.EqualError()` and `require.Equal()` vs their `assert.X` counterparts.

Reply:
got it thank you so much, it was `"github.com/stretchr/testify/require"`